src/libraries/Microsoft.PowerFx.Interpreter/Functions/Mutation/MutationUtils.cs

```diff
@@ -31,5 +33,38 @@ public static void CheckForReadOnlyFields(DType dataSourceType, TexlNode[] args,
                 }
             }
         }
+
+        /// <summary>
+        /// Merges all records starting from startIndex in args into a single record. Collisions are resolved by last-one-wins.
+        /// </summary>
+        /// <param name="records"></param>
+        /// <returns></returns>
+        public static DValue<RecordValue> MergeRecords(IEnumerable<FormulaValue> records)
+        {
+            var mergedFields = new Dictionary<string, FormulaValue>();
+
+            foreach (FormulaValue fv in records)
+            {
+                if (fv is ErrorValue errorValue)
+                {
+                    return DValue<RecordValue>.Of(errorValue);
+                }
+
+                if (fv is BlankValue)
+                {
+                    continue;
+                }
+
+                if (fv is RecordValue recordValue)
+                {
+                    foreach (var field in recordValue.Fields)
+                    {
+                        mergedFields[field.Name] = field.Value;
+                    }
+                }
+            }
```

Comment:
Suggestion (more compact):

    switch (fv)
    {
        case ErrorValue errorValue:
            return DValue<RecordValue>.Of(errorValue);
        case RecordValue recordValue:
            foreach (var field in recordValue.Fields)
            {
                mergedFields[field.Name] = field.Value;
            }

            break;
    }